src/vs/gitpod/browser/gitpodInsightsAppender.ts

```diff
@@ -81,33 +87,98 @@ export class GitpodInsightsAppender implements ITelemetryAppender {
-			}
-			if (!gitpodHost) {
+			const gitpodWsInfo = await this.getGitpodWorkspaceInfo();
+			if (!gitpodWsInfo.gitpodHost) {
 				return undefined;
 			}
 			// load grpc-web before see https://github.com/gitpod-io/gitpod/issues/4448
 			await import('@improbable-eng/grpc-web');
 			const { MetricsServiceClient, sendMetrics } = await import('@gitpod/ide-metrics-api-grpcweb');
-			const ideMetricsEndpoint = 'https://ide.' + gitpodHost + '/metrics-api';
+			const ideMetricsEndpoint = 'https://ide.' + gitpodWsInfo.gitpodHost + '/metrics-api';
 			const client = new MetricsServiceClient(ideMetricsEndpoint);
 			return async (metrics: IDEMetric[]) => {
 				await sendMetrics(client, metrics);
 			};
 		})();
 	}
 
+	private async sendErrorReports(error: ErrorEvent) {
+		const gitpodWsInfo = await this.getGitpodWorkspaceInfo();
+		const params: ReportErrorParam = {
+			workspaceId: gitpodWsInfo.workspaceId,
+			instanceId: gitpodWsInfo.instanceId,
+			errorStack: error.callstack,
+			userId: '',
+			component: 'vscode-workbench',
+			version: this._baseProperties.appName + '-' + this._baseProperties.version,  // Align to https://github.com/gitpod-io/gitpod/blob/fe26d2613df2f85667829c9de86bbeb187dc81cf/components/ide/code/leeway.Dockerfile#L75
+			properties: {
+				error_name: error.uncaught_error_name,
+				error_message: error.msg,
+				...this._baseProperties,
+			}
+		};
+		if (this.devMode && this.productService.gitpodPreview?.log?.errorReports) {
+			console.log('Gitpod Metrics: ', JSON.stringify(params, undefined, 2));
+		}
+		const doSend = await this.getSendErrorReports();
+		if (doSend) {
+			await doSend(params);
+		}
+	}
+
+	private _sendErrorReports: Promise<ErrorReports | undefined> | undefined;
+	private getSendErrorReports(): Promise<ErrorReports | undefined> {
+		if (this._sendErrorReports) {
+			return this._sendErrorReports;
+		}
+		return this._sendErrorReports = (async () => {
+			const gitpodWsInfo = await this.getGitpodWorkspaceInfo();
+			if (!gitpodWsInfo.gitpodHost) {
+				return undefined;
+			}
+			const ideMetricsHttpEndpoint = 'https://ide.' + gitpodWsInfo.gitpodHost + '/metrics-api/reportError';
+			return async (params: ReportErrorParam) => {
+				const response = await fetch(ideMetricsHttpEndpoint, {
+					method: 'POST',
+					body: JSON.stringify(params),
+					credentials: 'omit',
+				});
+				if (!response.ok) {
+					const data = await response.json();
+					console.error(`Cannot report error: ${response.status} ${response.statusText}`, data);
+				}
+			};
+		})();
+	}
+
+	private _gitpodWsInfo: Promise<SupervisorWorkspaceInfo> | undefined;
+	private getGitpodWorkspaceInfo(): Promise<SupervisorWorkspaceInfo> {
+		if (this._gitpodWsInfo) {
+			return this._gitpodWsInfo;
+		}
+		if (this.devMode) {
+			return this._gitpodWsInfo = (async () => ({
+				gitpodHost: this.productService.gitpodPreview?.host ?? 'gitpod-staging.com',
+				workspaceId: 'fake-workspace-id',
```

Comment:
we could still use supervisor to get real ids besides host? I think it is good to have some close to real data.